### PR TITLE
Modal window

### PR DESCRIPTION
**Pull request by VladiShcherbakov**
## Summary

- added modal window with additional information for recipes
- now API results is mapped into object before getting passed to the components
- data checks are moved to the mapping function 

### Screenshot

Bloody7_r30lrHTMPD

### List of steps to manually test introduced functionality:

- Go to http://localhost:3000
- Click on any recipe
- Click on either background or close button to close modal window

## Checklist

-   [x] My code follows the style guidelines of this project
-   [ ] I have added tests for new functionality
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] My pull-request is shorter than 500 lines

### CODE REVIEW

File: src/components/ModalWindow/ModalWindow.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { ModalBackground, ModalContainer, ModalImage } from "./styles";
+
+const ModalWindow = ({ showModal, customOnClick, modalParent }) => {
+  // eslint-disable-next-line no-console
+  console.log(modalParent);
+  return (
+    <ModalBackground showModal={showModal} onClick={customOnClick}>
+      <ModalContainer onClick={(e) => e.stopPropagation()}>
+        <h2>Test</h2>
```
Comment (nadezhdakharchuk) [inline src/components/ModalWindow/ModalWindow.jsx:10]:
```
<ModalContainer onClick={(e) => e.stopPropagation()}>
   {children}
</ModalContainer>
```

File: src/components/RecipesList/RecipesList.jsx [hunk 2]
```diff
@@ -28,13 +36,19 @@ const RecipesList = () => {
 
   return (
     <List>
+      <ModalWindow
```
Comment (nadezhdakharchuk) [inline src/components/RecipesList/RecipesList.jsx:39]: <ModalWindow>
  <CustomComponent receipt={receipt} />
</ModalWindow>

File: src/components/RecipesList/RecipesList.jsx [hunk 3]
```diff
@@ -28,13 +36,19 @@ const RecipesList = () => {
...
+        showModal={showModal}
+        customOnClick={() => setShowModal(false)}
+        modalInfo={modalInfo}
+      />
       {recipes.map(({ recipe }) => (
         <RecipeItem
           key={recipe.label}
```
Comment (nadezhdakharchuk) [inline src/components/RecipesList/RecipesList.jsx:46]: `key={recipe.id}`

File: src/components/RecipeItemModal/RecipeItemModal.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+const RecipeModalWindow = ({ recipeData, customOnClick }) => {
+  const totalWeight = recipeData?.totalWeight ? recipeData.totalWeight : 1;
+  const KCAL = recipeData?.nutrients.ENERC_KCAL.quantity
+    ? recipeData.nutrients.ENERC_KCAL.quantity
+    : 0;
+
+  return (
+    <ModalContainer onClick={(e) => e.stopPropagation()}>
+      <ButtonWrapper>
```
Comment (nadezhdakharchuk) [inline src/components/RecipeItemModal/RecipeItemModal.jsx:22]: Could you please move close button to modal window component and fix scrolling issues?

https://user-images.githubusercontent.com/14818162/206129039-21f1a468-b760-420e-8076-5c626c19ca7f.mp4

File: src/components/RecipesList/RecipesList.jsx [hunk 5]
```diff
@@ -1,16 +1,72 @@
...
+
+  const setModalData = (productData) => {
+    setRecipeID(productData);
+    setShowModal(true);
+  };
+
+  const mapApiResult = (result) => {
+    const mappedResult = result.map(({ recipe }) => ({
+      id: uuidv4(),
+      label: recipe.label,
```
Comment (nadezhdakharchuk) [inline src/components/RecipesList/RecipesList.jsx:22]: This data we can add just like
```
const mappedResult = result.map(({ recipe }) => ({
      id: uuidv4(),
      ...recipe
})

```

and prepare all info about recipe inside the RecipeItem

**Review by NadezhdaShosheva — APPROVED** [review]

**Pull request merged**